File: http/http-server-annotation-processor/src/test/java/ru/tinkoff/kora/http/server/annotation/processor/controller/TestControllerWithInheritance.java
package ru.tinkoff.kora.http.server.annotation.processor.controller;

import ru.tinkoff.kora.http.common.HttpMethod;
import ru.tinkoff.kora.http.common.annotation.HttpRoute;
import ru.tinkoff.kora.http.server.common.annotation.HttpController;

@HttpController("/base")
public class TestControllerWithInheritance extends ParentTestController<String> {

    @HttpRoute(method = HttpMethod.GET, path = "/child")
    public String someOtherMethod() {
        return "child";
    }
}

File: http/http-server-symbol-processor/src/test/kotlin/ru/tinkoff/kora/http/server/symbol/processor/controllers/TestControllerWithInheritance.kt
package ru.tinkoff.kora.http.server.symbol.processor.controllers

import ru.tinkoff.kora.http.common.HttpMethod
import ru.tinkoff.kora.http.common.annotation.HttpRoute
import ru.tinkoff.kora.http.server.common.annotation.HttpController

@HttpController("/base")
open class TestControllerWithInheritance : ParentTestController<String?>() {
    @HttpRoute(method = HttpMethod.GET, path = "/child")
    open fun someOtherMethod(): String {
        return "child"
    }
}
